libs/community/langchain_community/vectorstores/oraclevs.py

```diff
@@ -401,6 +417,39 @@ def __init__(
                 "`pip install -U oracledb`."
             ) from e
 
+        self.insert_mode = "array"
+
+        if client.thin is True:
+            if oracledb.__version__ == "2.1.0":
+                raise Exception(
+                    "Oracle DB python thin client driver version 2.1.0 not supported"
+                )
+            elif _compare_version(oracledb.__version__, "2.2.0"):
+                self.insert_mode = "clob"
+            else:
+                self.insert_mode = "array"
+        else:
+            if (_compare_version(oracledb.__version__, "2.1.0")) and (
+                not (
+                    _compare_version(
+                        ".".join(map(str, oracledb.clientversion())), "23.4"
+                    )
+                )
+            ):
+                raise Exception(
+                    "Oracle DB python thick client driver version earlier than "
+                    "2.1.0 not supported with client libraries greater than "
+                    "equal to 23.4"
+                )
+
+            if _compare_version(".".join(map(str, oracledb.clientversion())), "23.4"):
+                self.insert_mode = "clob"
+            else:
+                self.insert_mode = "array"
+
+            if _compare_version(oracledb.__version__, "2.1.0"):
+                self.insert_mode = "clob"
```

Comment:
Could you combine this into one if/else block with an OR

Reply:
We wanted to keep the logic like our internal docs. Can we please keep it like this